### PR TITLE
feat: watch version status

### PR DESCRIPTION
**Pull request by guumaster**

### CODE REVIEW

File: admin/admin-ui/src/Components/SubscriptionService/Services/onVersionStatusChanged/onVersionStatusChanged.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import { loader } from 'graphql.macro';
+import { WatchVersionStatus } from 'Graphql/subscriptions/types/WatchVersionStatus';
+import { get } from 'lodash';
+import ROUTE from '../../../../Constants/routes';
+
+const VersionStatusSubscription = loader(
+  'Graphql/subscriptions/watchVersionStatus.graphql'
+);
+
+function onVersionStatus(client: ApolloClient<object>) {
```
Comment (mgs95) [inline admin/admin-ui/src/Components/SubscriptionService/Services/onVersionStatusChanged/onVersionStatusChanged.tsx:11]: exported function is ```onVersionStatus``` and file is ```onVersionStatusChanged```, both should be ```onVersionStatus```

File: admin/admin-ui/src/Components/SubscriptionService/Services/onVersionStatusChanged/onVersionStatusChanged.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  console.log('starting onVersionStatus subscription');
+  client
+    .subscribe<WatchVersionStatus>({
+      query: VersionStatusSubscription
+    })
+    .subscribe({
+      next(data) {
+        console.log('SUBSCRIPTION DATA: ', data);
```
Comment (mgs95) [inline admin/admin-ui/src/Components/SubscriptionService/Services/onVersionStatusChanged/onVersionStatusChanged.tsx:19]: Remove this log. does the subscription works without defining this function? if not we can set the content of the function empty

File: admin/admin-api/adapter/gql/resolver.go [hunk 3]
```diff
@@ -97,12 +99,54 @@ func (r *mutationResolver) CreateVersion(ctx context.Context, input CreateVersio
...
+
+	go func() {
+		for {
+			select {
+			case v, ok := <-notifyCh:
+				if !ok {
+					return
+				}
+
+				for _, r := range versionStatusChannels {
```
Comment (angelsanchez) [inline admin/admin-api/adapter/gql/resolver.go:116]: why a versionStatusChannels is it called `r`?

File: admin/admin-api/adapter/gql/resolver.go [hunk 4]
```diff
@@ -97,12 +99,54 @@ func (r *mutationResolver) CreateVersion(ctx context.Context, input CreateVersio
...
+					r <- v
+				}
+			}
+		}
+	}()
```
Comment (angelsanchez) [inline admin/admin-api/adapter/gql/resolver.go:147]: duplicated code

File: admin/admin-api/domain/usecase/version_interactor.go [hunk 5]
```diff
@@ -250,87 +250,140 @@ func (i *VersionInteractor) generateWorkflows(krtYml *krt.Krt) ([]*entity.Workfl
...
+		defer close(notifyStatusCh)
+		err = i.versionService.Start(ctx, runtime, v)
+		if err != nil {
+			// TODO: Should we handle ERROR status?
+			i.logger.Errorf("[versionInteractor.Start] error starting version '%s': %s", v.Name, err)
+		}
+
+		err = i.versionRepo.SetStatus(ctx, v.ID, entity.VersionStatusStarted)
+		if err != nil {
+			// TODO: Should we handle ERROR status?
```
Comment (angelsanchez) [inline admin/admin-api/domain/usecase/version_interactor.go:311]: I think printing the error into the logs is enough

File: admin/admin-api/domain/usecase/version_interactor.go [hunk 6]
```diff
@@ -250,87 +250,140 @@ func (i *VersionInteractor) generateWorkflows(krtYml *krt.Krt) ([]*entity.Workfl
...
 	}
-	return v, nil
+
+	go func() {
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Minute) // TODO: consider to move version starting timeout to config
+		defer cancel()
+		defer close(notifyStatusCh)
+		err = i.versionService.Start(ctx, runtime, v)
+		if err != nil {
+			// TODO: Should we handle ERROR status?
```
Comment (angelsanchez) [inline admin/admin-api/domain/usecase/version_interactor.go:305]: I think printing the error into the logs is enough

File: admin/admin-api/domain/usecase/version_interactor.go [hunk 7]
```diff
@@ -250,87 +250,140 @@ func (i *VersionInteractor) generateWorkflows(krtYml *krt.Krt) ([]*entity.Workfl
...
 
 	err = i.userActivityInteractor.RegisterStopAction(loggedUserID, runtime, v, comment)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
-	return v, nil
+
+	go func() {
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Minute) // TODO: consider to move version starting timeout to config
```
Comment (angelsanchez) [inline admin/admin-api/domain/usecase/version_interactor.go:367]: stopping timeout

File: admin/admin-api/adapter/gql/resolver.go (hunk 3 above)
Comment (guumaster) [reply admin/admin-api/adapter/gql/resolver.go]: fixed

File: admin/admin-api/adapter/gql/resolver.go (hunk 4 above)
Comment (guumaster) [reply admin/admin-api/adapter/gql/resolver.go]: de-duplicated

File: admin/admin-api/domain/usecase/version_interactor.go (hunk 7 above)
Comment (guumaster) [reply admin/admin-api/domain/usecase/version_interactor.go]: moved to config

**Pull request merged**